### PR TITLE
fix(impact): pass the IMPACT models their statistics flat and in itk-impact's order

### PR DESCRIPTION
**Pull request by vboussot**
## Problem

`ImpactFeatureModel.inputs` built the intensity statistics as `[[min, mean, max, std]]`. The IMPACT TorchScript models read `stats` only when it holds exactly 4 values, and expect it flat, in itk-impact's order `[min, max, mean, std]` (`itkImpactModelConfigurationDetail.h`). Otherwise they compute the statistics from `x` themselves.

Read out of the shipped models (`model.normalize.code`):

| model | with 4 values | otherwise |
|---|---|---|
| MIND | `(x - stats[0]) / (stats[1] - stats[0])` | min/max of the tensor it is handed |
| TS MRI (M730/M850/M852) | `(x - stats[2]) / stats[3]` | mean/std of the tensor it is handed |
| SAM 2.1 | same as MIND, then ImageNet mean/std | min/max of the tensor it is handed |
| TS CT (M291), Anatomix | ignore `stats` | fixed buffers |

Three bugs followed:

- **Crash with a single sample.** The `[1, 4]` row also has `numel() == 4`, so MIND and TS took the branch that reads `stats`. `stats[0]` was then a 4-vector subtracted from the image, and MIND failed with `size of tensor a (32) must match the size of tensor b (4)`. This broke every FireANTs + IMPACT/MIND run.
- **Wrong normalisation.** Because the order was wrong, MIND divided by `(mean - min)` instead of `(max - min)`, and TS MRI standardised around the max instead of the mean.
- **Batch-wide normalisation.** A `[B, 4]` block is not 4 values either, so every sample of a batch was normalised by the *batch's* min/max or mean/std, which is another case's intensities. SAM is worse: its gate is `stats.size(0) == 4`, so a batch of exactly 4 passed it and took a whole row as the minimum.

impact-reg already had a fix: `_ImpactCore.preprocessing` flattened the row. It became dead code when the measures moved to `ImpactFeatureModel.inputs`. Nothing calls it any more, and the `super().preprocessing` it calls no longer exists.

## Change

- `inputs` takes one sample's attribute and returns itk-impact's flat order.
- `slice_losses` scores one sample at a time, as itk-impact does and as `_pca_project` already did for the same reason. Same work in more kernels: **+5 % (B=2) to +17 % (B=4)** of the model's time, measured on an RTX PRO 5000.
- The dead override is removed from impact-reg.
- Two regression tests: the order, and each sample of a batch reading its own row.

Checked against the real MIND model, on a batch pairing a CT-scale volume with a `[0, 1]` one: the batch now scores `0.287980`, exactly the mean of the two scored alone, where it scored `0.144727` before.

## Behaviour change

A single sample used to crash before it could read the wrong order, so nothing that worked is changed there. **A batch of B > 1 now scores differently**: it used to be normalised batch-wide, it is now normalised per sample. Runs whose samples share an intensity scale barely move; a mixed batch moves a lot (2x above, deliberately extreme).

## Summary by CodeRabbit

- **Bug Fixes**
  - Corrected image statistics ordering to consistently report minimum, maximum, mean, and standard deviation.
  - Preserved batch-shaped statistics for multi-sample inputs while maintaining appropriate formatting for single-sample inputs.
  - Ensured batched scoring uses the corresponding statistics and masks for each sample and applicable slice.
  - Prevented preprocessing from unintentionally flattening singleton-batch statistics.

- **Tests**
  - Added regression coverage for statistics ordering and batch dimensions.

### CODE REVIEW

File: konfai/metric/measure/impact.py [hunk 1]
```diff
@@ -220,13 +220,14 @@ def download(
...
+        ``[ImageMin, ImageMax, ImageMean, ImageStd]`` row per sample, flat for a single sample: the order
+        itk-impact passes, and the models only read ``stats`` when ``stats.numel() == 4``."""
         if tensor.shape[1] != self.in_channels:
             tensor = tensor.repeat(1, self.in_channels, *([1] * (tensor.dim() - 2)))
         if self.denormalize:
             tensor = _denormalized(tensor, attributes)
-        stats = [[float(a[key]) for key in ("ImageMin", "ImageMean", "ImageMax", "ImageStd")] for a in attributes]
-        return [tensor, torch.tensor([len(self.weights)]), torch.tensor(stats)]
+        stats = [[float(a[key]) for key in ("ImageMin", "ImageMax", "ImageMean", "ImageStd")] for a in attributes]
+        return [tensor, torch.tensor([len(self.weights)]), torch.tensor(stats).squeeze(0)]
```
[the remark this answers is not in the document]
Comment (vboussot) [reply konfai/metric/measure/impact.py]: Confirmed and fixed in 744b4e6.

Read out of the shipped TorchScript models rather than inferred: MIND falls back to `torch.min(x)` / `torch.max(x)` over the whole tensor, TS MRI to `torch.mean(x)` / `torch.std(x)` over the whole tensor. With a batch that is another case's intensities, exactly as you describe. SAM 2.1 is worse than the fallback: its gate is `stats.size(0) == 4`, not `numel()`, so a batch of exactly 4 passed it and took a whole 4-value row as the minimum.

`slice_losses` now scores one sample at a time, which is what itk-impact does and what `_pca_project` in this same file already did for the same reason. Splitting the inference was the option to take: the alternative, teaching every published TorchScript model to accept `[B, 4]`, is not ours to do, the models are traced and shipped.

Cost: the same work in more kernels, measured on an RTX PRO 5000 at 96^3, +5 % at B=2 and +17 % at B=4 of the model's time.

Regression test added (`test_impact_scores_a_batch_one_sample_at_a_time_with_its_own_statistics`): a recording model asserts each sample is handed its own flat row. Checked against the real MIND model too, on a batch pairing a CT-scale volume with a `[0, 1]` one: the batch now scores 0.287980, exactly the mean of the two scored alone, against 0.144727 before.
